### DISCUSSION TITLE
s/regex/chatgpt/

### DISCUSSION THREAD

**Discussion by phips** [ai-robots-txt/ai.robots.txt#34] [Show and tell]
I used to love writing regexs. `man perlre` is one's friend. However, these days I count myself as a 'retired engineer', and I'm either slow or, being candid, can't be bothered. I keep this super helpful blocklist in an nginx config (thanks to Robb Knight's blog for that), but when it's updated I just want to swap out the list.

I thought for a second about doing a quick macro in Vim, then I thought I could, should, code something quick and dirty and maybe submit to the repo. Then I thought _"I wonder if ChatGPT can sort this?"_

So I tried:

`Hey! Can you take this list and remove "User-agent:\s" then join every entry together with | ?`

```
Here’s the list with “User-agent: “ removed and all entries joined together with |:

AI2Bot|Ai2Bot-Dolma|Amazonbot|Applebot|Applebot-Extended|Bytespider|CCBot|ChatGPT-User|
Claude-Web|ClaudeBot|Diffbot|FacebookBot|FriendlyCrawler|GPTBot|Google-Extended|
GoogleOther|GoogleOther-Image|GoogleOther-Video|ICC-Crawler|ImagesiftBot|Meta-ExternalAgent|
Meta-ExternalFetcher|OAI-SearchBot|PerplexityBot|PetalBot|Scrapy|Timpibot|VelenPublicWebCrawler|
Webzio-Extended|YouBot|anthropic-ai|cohere-ai|facebookexternalhit|img2dataset|omgili|omgilibot
```

Sorted 😄 

There’s a delicious irony in asking chatgpt to sort a blocklist of AI content scrapers for nginx, don’t you think?

**Comment by glyn** [comment]
Nice. I wonder if it could write a correct shell script to perform the same transformation?

**Reply by phips** [reply]
I hate myself for doing this, but… 🤣 

![CleanShot 2024-08-29 at 20 28 00@2x](https://github.com/user-attachments/assets/6ebcefec-314a-461c-b477-aab515467234)
![CleanShot 2024-08-29 at 20 28 13@2x](https://github.com/user-attachments/assets/6e987820-45ab-4513-9057-5492f827c178)
![CleanShot 2024-08-29 at 20 28 25@2x](https://github.com/user-attachments/assets/242fb511-a38c-4773-8e52-c1842099837e)

*shudder* So dirty, so, so dirty:

![CleanShot 2024-08-29 at 20 29 20@2x](https://github.com/user-attachments/assets/48ca6cad-68a6-4635-aa7d-52bfc81f5582)


And for copy/paste use:

`awk '{gsub(/^User-agent: /, ""); s = s $0 "|"} END {sub(/\|$/, "", s); print s}' user_agents.txt`
